### ISSUE TITLE
Conditional requiredIf breaking on change

### ISSUE THREAD

**Issue by opheliadesign** [kazupon/vue-validator#56]
See http://stackoverflow.com/questions/32641467/conditional-validation-in-vuejs for details of problem and code. Essentially, I am trying to conditionally require different fields based on a boolean value. Validation seems to break if the boolean is changed from false to true and back to false.

If `someField` is true, I want `fieldA` to be required. If `someField` is false, I want `fieldA` to  NOT be required, `fieldB` and `fieldC` to be required.

**Comment by opheliadesign** [comment]
I'd really appreciate some thoughts on this. Please let me know if further information is required. Thanks!

**Comment by kazupon** [comment]
Sorry for late reply.
I checked your stackoverflow code.
Sorry, I cannot understand this essential issue.

Would you provide a your reporting code with jsfiddle ?
That time, I would appreciate if you provide a reproducible to me.

**Comment by opheliadesign** [comment]
I honestly don't know how else to explain it. If a condition (`observerEntry`) is true, two fields (`shiftEntry.observerFirstName`, `shiftEntry.observerLastName`) need to be required while a third field (`shiftEntry.member`) is not required. If `observerEntry` is false, `shiftEntry.observerFirstName` and `shiftEntry.observerLastName` are NOT required and `shiftEntry.member` IS required. So. all three fields need to be `required_if` based on the boolean `observerShiftEntry`.

I was able to solve this not using the validation library but I'd like to see if it's possible using vuejs-validator. Here is a jsfiddle with the code NOT using your library to show the desired result - https://jsfiddle.net/6gpryhep/

If the  Role is anything but blank or "Observer," the form will validate when a "Member" is selected. If it is set to "Observer," the form will validate after First Name and Last Name have been entered. This is based on the computed boolean of `observerShift`. So, again, First Name and Last Name are "required if" `observerShift` is true.

**Comment by kazupon** [comment]
Thank you for your jsfiddle code !!

I tried to  reproduce you have a problem.
https://jsfiddle.net/kazupon/gqLdjL3d/5/

Is this right ?

**Comment by opheliadesign** [comment]
The desired result is shown in the jsfiddle I provided. If the role is anything but empty or "Observer" and a Member is selected, the form validates successfully. If "Observer" is set but the First Name and Last Name fields are not entered, the form is invalid. 

In your example, the form is only valid if the Member is selected (so Role anything other than empty or Observer) AND the First Name and Last Name fields are not empty. The First Name and Last Name fields should only be required _if_ Observer is not set. Likewise, Member should only be required _if_ Role is not set to Observer.

Also, your fiddle is displaying the same behavior I noticed previously - when the page loads, the Member field is blank and highlighted in red indicating the error. If `showObserver` is set to `true` and then back to `false`, validation on the `member` field seems to break and the validation object becomes empty `{ }`
